rfd/0123-tls-routing-behind-layer7-lb.md

```diff
@@ -299,15 +323,56 @@ for different clusters.
 UX is the same as when [TLS Routing is
 enabled](https://github.com/gravitational/teleport/blob/master/rfd/0039-sni-alpn-teleport-proxy-routing.md#local-teleport-proxy).
 
+#### 5 - Client source IPs
+
+A new setting is added to the Proxy service for web APIs to take client IPs from the "X-Forwarded-For" headers:
+```
+proxy_service:
```

Comment:
What do you think about doing a  WARN check during teleport proxy startup if based to determine if proxy is behind L7 LB and `use_xff_header` needs to be enabled to support proper client IP forwarding.

Reply:
It's unclear to me if the check needs to be so complicated.  I think a warning if the option is simply enabled (or once the service realizes it's enabled) is sufficient.  Even if the proxy does not send the XFF header, there is the potential a malicious client may send this header.  As long as the option is on to inspect for the header there is a potential risk.

Let me know if I missed anything!

Reply:
@smallinsky correct me if I am wrong, I believe the warning is for when the option is **OFF** + we detect Proxy is behind L7 LB, a message can be shown to user that ips in log and IP pinning feature may not function properly with the option disabled, and we can provide the user a guide on how to enable it.

Personally, I don't think we need a warning when the option is ON, since whoever configuring it should understand the risk. We will document this properly on the guide/references.

Reply:
Thank you for the clarification @greedy52, it looks like I was confused.  I was thinking this warning would be to ensure customers understood the security risks.  I don't view this as a requirement, but to me there seems to be very little value in warning when it's disabled.